File: app/src/main/java/com/mkrworld/libfilter/filter/overlayfilter/BaseOverlayFilter.java
package com.mkrworld.libfilter.filter.overlayfilter;

import android.util.Log;

import com.mkrworld.libfilter.dto.FilterMatrix;
import com.mkrworld.libfilter.enums.FilterCategory;
import com.mkrworld.libfilter.filter.BaseFilter;
import com.mkrworld.libfilter.jnicaller.Effector;

import java.util.ArrayList;

public abstract class BaseOverlayFilter extends BaseFilter {

    private int[] mOverlayPixelArray;
    private float mMultiplier;

    /**
     * Constructor
     *
     * @param pixelArray        Array of Pixel
     * @param imageWidth
     * @param overlayPixelArray
     * @param multiplier
     */
    public BaseOverlayFilter(int[] pixelArray, int imageWidth, int[] overlayPixelArray, float multiplier) {
        super(pixelArray, imageWidth);
        mOverlayPixelArray = overlayPixelArray;
        mMultiplier = multiplier;
    }

    /**
     * Method to get the Multiplier
     * @return
     */
    public float getMultiplier() {
        return mMultiplier;
    }

    /**
     * Method to get the Overlay Image Pixel Array
     * @return
     */
    public int[] getOverlayPixelArray() {
        return mOverlayPixelArray;
    }
}

File: app/src/main/java/com/mkrworld/libfilter/filter/BaseOverlayFilter.kt
package com.mkrworld.libfilter.filter

/**
 * @author THEMKR
 */
abstract class BaseOverlayFilter : BaseFilter {

    /**
     * Method to get the Overlay Image Pixel Array
     * @return
     */
    val overlayPixelArray: IntArray
    /**
     * Method to get the Multiplier
     * @return
     */
    val multiplier: Float

    /**
     * Constructor
     *
     * @param pixelArray        Array of Pixel
     * @param imageWidth
     * @param overlayPixelArray
     * @param multiplier
     */
    constructor(pixelArray: IntArray, imageWidth: Int, overlayPixelArray: IntArray, multiplier: Float) : super(pixelArray, imageWidth, arrayListOf()) {
        this.overlayPixelArray = overlayPixelArray
        this.multiplier = multiplier
    }
}
